### ISSUE TITLE
Missing  logger output using tensor board

### ISSUE THREAD

**Issue by tanyaapaul** [DLR-RM/stable-baselines3#1279]
### ❓ Question

Hello, 

I am currently using Tensorboard to output the result of the DQN, however I note in the output I am only receiving the
rollout and the time metrics but no evaluation neither training metrics. Is there a way to log more values/ metrics into the output? Is it related to the fact that I am using DQN, as it is also said that the output of the logs varies based on the model chosen. Any tips would be helpful please :) 

 I saw in the documentation it suggests : 

1) 'If you want to log values more often than the default to tensorboard, you manually call self.logger.dump(self.num_timesteps) in a callback (see [issue #506](https://github.com/DLR-RM/stable-baselines3/issues/506)).'

2) Or to use callback=TensorboardCallback())

Currently I am using eval call back : 

```
callbacks = []
eval_callback = EvalCallback(
    env,
    callback_on_new_best=None,
    n_eval_episodes=5,
    best_model_save_path=".",
    log_path=".",
    eval_freq=10000,
)
callbacks.append(eval_callback)

```
 

Currently I only have those outputs 
```
 rollout/                |             |
|    ep_len_mean          | 200         |
|    ep_rew_mean          | -227        |
| time/                   |             |
|    fps                  | 972         |
|    iterations           | 19          |
|    time_elapsed         | 80          |
|    total_timesteps      | 77824       |




```

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] If code there is, it is minimal and working
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Hello,

>  If code there is, it is minimal and working

please fill the issue template completely (current code is minimal but not working, missing imports and statements)

Two ways to have evaluation reported:

```python
from stable_baselines3 import DQN
from stable_baselines3.common.env_util import make_vec_env
from stable_baselines3.common.callbacks import EvalCallback

env_id = "CartPole-v1"

eval_env = make_vec_env(env_id, n_envs=2)
callback = EvalCallback(eval_env, eval_freq=10000)

model = DQN(
    "MlpPolicy",
    env_id,
    learning_starts=1000,
    verbose=1,
    tensorboard_log="/tmp/sb3",
)

model.learn(int(1e5), log_interval=100, callback=callback, progress_bar=True)
```

or using the RL Zoo:
```
python3 -m rl_zoo3.train --algo dqn --env CartPole-v1 --eval-freq 10000 -tb /tmp/sb3/ --log-interval 100 -P
```

every 1000 steps, it will display:
```
Eval num_timesteps=20000, episode_reward=144.60 +/- 4.76
Episode length: 144.60 +/- 4.76
----------------------------------
| eval/               |          |
|    mean_ep_length   | 145      |
|    mean_reward      | 145      |
| rollout/            |          |
|    exploration_rate | 0.04     |
| time/               |          |
|    total_timesteps  | 20000    |
| train/              |          |
|    learning_rate    | 0.0023   |
|    loss             | 0.41     |
|    n_updates        | 9600     |
----------------------------------

```

> Currently I only have those outputs

It seems you are not using DQN (`exploration_rate` is missing)

**Comment by tanyaapaul** [comment]
Hello ! 

Sorry my bad I am using airtime to run this DQN here is the entire code 👍 
```
import setup_path
import gym
import airgym
import time

from stable_baselines3 import DQN
from stable_baselines3.common.monitor import Monitor
from stable_baselines3.common.vec_env import DummyVecEnv, VecTransposeImage
from stable_baselines3.common.evaluation import evaluate_policy
from stable_baselines3.common.callbacks import EvalCallback

# Create a DummyVecEnv for main airsim gym env
env = DummyVecEnv(
    [
        lambda: Monitor(
            gym.make(
                "airgym:airsim-drone-sample-v0",
                ip_address="127.0.0.1",
                step_length=0.25,
                image_shape=(84, 84, 1),
            )
        )
    ]
)

# Wrap env as VecTransposeImage to allow SB to handle frame observations
env = VecTransposeImage(env)

# Initialize RL algorithm type and parameters
model = DQN(
    "CnnPolicy",
    env,
    learning_rate=0.00025,
    verbose=1,
    batch_size=32,
    train_freq=4,
    target_update_interval=10000,
    learning_starts=10000,
    buffer_size=500000,
    max_grad_norm=10,
    exploration_fraction=0.1,
    exploration_final_eps=0.01,
    device="cuda",
    tensorboard_log="./tb_logs/",
)

# Create an evaluation callback with the same env, called every 10000 iterations
callbacks = []
eval_callback = EvalCallback(
    env,
    callback_on_new_best=None,
    n_eval_episodes=5,
    best_model_save_path=".",
    log_path=".",
    eval_freq=10000,
)
callbacks.append(eval_callback)

kwargs = {}
kwargs["callback"] = callbacks

# Train for a certain number of timesteps
model.learn(
    total_timesteps=5e5,
    tb_log_name="dqn_airsim_drone_run_" + str(time.time()),
    **kwargs
)

# Save policy weights
model.save("dqn_airsim_drone_policy")


```
Currently I only have as log output : 
- exploration rate 
- 2 rollout outputs 
-  all the time outputs 
but missing evaluation and train 

I am comparing it to the doc here : https://stable-baselines3.readthedocs.io/en/master/common/logger.html and notice there are a lot more possible outputs which are not generated in my log outputs

**Comment by araffin** [comment]
> Currently I only have as log output :

can you confirm that it does work with the code I provided?

**Comment by tanyaapaul** [comment]
I am trying to integrate your suggestions into my code, but not sure how. Are you suggesting to change my model.learn and add  callback=callback, progress_bar=True ?

**Comment by araffin** [comment]
> can you confirm that it does work with the code I provided?

I'm just asking if the code I provide does work on your side.

Callback as a list or single element should be equivalent, `progress_bar=True` is just to have more feedback, not needed.

**Comment by tanyaapaul** [comment]
Ok thanks for your help! I will test it out, but assuming it works , how will it help adding the outputs missing in my DQN logger?

**Comment by araffin** [comment]
> how will it help adding the outputs missing in my DQN logger?

well, you will have a working code, so you can progressively change it until it fits your needs.
And by looking at the differences, you might be able to find what's wrong.

**Comment by tanyaapaul** [comment]
Ok thanks! will do this then!

**Comment by Geoffreyvd1** [comment]
@araffin  why is this not default for DQN. I think having the train loss in there as default makes a lot of sense
